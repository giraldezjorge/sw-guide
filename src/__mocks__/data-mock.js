
export const mockCharacter = {
  name: 'Luke Skywalker',
  height: '172',
  mass: '77',
  hair_color: 'blond',
  skin_color: 'fair',
  eye_color: 'blue',
  birth_year: '19BBY',
  gender: 'male',
  url: 'test/people/1/'
}

export const mockCharacters = [
  mockCharacter,
  {
    name: 'C-3PO',
    height: '167',
    mass: '75',
    hair_color: 'n/a',
    skin_color: 'gold',
    eye_color: 'yellow',
    birth_year: '112BBY',
    gender: 'n/a',
    url: 'test/people/2/'
  },
  {
    name: 'R2-D2',
    height: '96',
    mass: '32',
    hair_color: 'n/a',
    skin_color: 'white, blue',
    eye_color: 'red',
    birth_year: '33BBY',
    gender: 'n/a',
    url: 'test/people/3/'
  }
]

export const mockFilm = {
  title: 'A New Hope',
  episode_id: 4,
  opening_crawl: 'It is a period of civil war.\r\nRebel spaceships, striking\r\nfrom a hidden base, have won\r\ntheir first victory against\r\nthe evil Galactic Empire.\r\n\r\nDuring the battle, Rebel\r\nspies managed to steal secret\r\nplans to the Empire\'s\r\nultimate weapon, the DEATH\r\nSTAR, an armored space\r\nstation with enough power\r\nto destroy an entire planet.\r\n\r\nPursued by the Empire\'s\r\nsinister agents, Princess\r\nLeia races home aboard her\r\nstarship, custodian of the\r\nstolen plans that can save her\r\npeople and restore\r\nfreedom to the galaxy....',
  director: 'George Lucas',
  producer: 'Gary Kurtz, Rick McCallum',
  release_date: '1977-05-25',
  url: 'test/films/1/',
  characters: [
    'test/people/1/'
  ]
}

export const mockFilms = [
  mockFilm,
  {
    title: 'The Empire Strikes Back',
    episode_id: 5,
    opening_crawl: 'It is a dark time for the\r\nRebellion. Although the Death\r\nStar has been destroyed,\r\nImperial troops have driven the\r\nRebel forces from their hidden\r\nbase and pursued them across\r\nthe galaxy.\r\n\r\nEvading the dreaded Imperial\r\nStarfleet, a group of freedom\r\nfighters led by Luke Skywalker\r\nhas established a new secret\r\nbase on the remote ice world\r\nof Hoth.\r\n\r\nThe evil lord Darth Vader,\r\nobsessed with finding young\r\nSkywalker, has dispatched\r\nthousands of remote probes into\r\nthe far reaches of space....',
    director: 'Irvin Kershner',
    producer: 'Gary Kurtz, Rick McCallum',
    release_date: '1980-05-17',
    url: 'test/films/2/',
    characters: [
      'test/people/1/'
    ]
  }
]
