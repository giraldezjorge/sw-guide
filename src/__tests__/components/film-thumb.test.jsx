import React from 'react'
import { MemoryRouter } from 'react-router-dom'
import { render, mount, shallow } from 'enzyme'
import renderer from 'react-test-renderer'
import { mockFilm } from '../../__mocks__/data-mock'
import FilmThumb from '../../components/film-thumb'

describe('Film thumb', () => {
  const film = mockFilm

  it('renders correctly', () => {
    const component = renderer.create(
      <MemoryRouter>
        <FilmThumb film={film} />
      </MemoryRouter>
    )
    const tree = component.toJSON()
    expect(tree).toMatchSnapshot()
  }),

  it('should contain the film episode', () => {
    const component = mount(
      <MemoryRouter>
        <FilmThumb film={film} />
      </MemoryRouter>
    )

    expect(component.find('.card-content .media .media-content p.subtitle').text()).toEqual('Episode#4')
  })
})
