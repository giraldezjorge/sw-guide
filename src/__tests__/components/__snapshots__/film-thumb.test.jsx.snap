// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Film thumb renders correctly 1`] = `
<div
  className="sc-bdVaJa fVCJpx card"
  onClick={[Function]}
>
  <div
    className="card-image"
  >
    <figure
      className="image is-4by3"
    >
      <img
        alt="Thumbnail"
        src="sw_5.jpg"
      />
    </figure>
  </div>
  <div
    className="card-content"
  >
    <div
      className="media"
    >
      <div
        className="media-content"
      >
        <p
          className="subtitle is-6"
        >
          Episode#4
        </p>
        <p
          className="title is-4"
        >
          A New Hope
        </p>
      </div>
    </div>
  </div>
</div>
`;
